flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/SetOperationsITCase.scala

```diff
@@ -139,4 +154,105 @@ class UnionITCase(
     // Must fail. Tables are bound to different TableEnvironments.
     ds1.unionAll(ds2).select('c)
   }
+
+  @Test
+  def testSetMinusAll(): Unit = {
```

Comment:
Can you combine this and the next test by using test data that covers both cases for different records, i.e., have some records with duplicates in this first, second, none, and both data sets.